src/extractor/guidance/node_based_graph_walker.cpp

```diff
@@ -48,6 +51,7 @@ void LengthLimitedCoordinateAccumulator::update(const NodeID from_node,
     coordinates.insert(coordinates.end(), current_coordinates.begin(), current_coordinates.end());
 
     accumulated_length += length;
+    accumulated_length = std::min(accumulated_length, max_length);
```

Comment:
hm. the coordinates are trimmed to the requested length as well. It essentially gives the information is the distance could be walked completely.